src/send_test_notification.py

```diff
@@ -0,0 +1,171 @@
+    if build_context == "PR":
+        return os.getenv("DLC_IMAGES")
+    if build_context == "MAINLINE":
+        test_env_file = os.path.join(
+            os.getenv("CODEBUILD_SRC_DIR_DLC_IMAGES_JSON"), "test_type_images.json"
+        )
+        with open(test_env_file) as test_env:
+            test_images = json.load(test_env)
+        for dlc_test_type, images in test_images.items():
+            if dlc_test_type == "sanity":
+                return " ".join(images)
+        raise RuntimeError(f"Cannot find any images for in {test_images}")
+
+
+def get_platform_execution_details(build_context):
+    platform_details = {}
+    codebuild_name = get_codebuild_project_name()
+
+    platform_details["platform_info"] = {}
+    platform_details["platform_info"]["build_context"] = build_context
+    platform_details["platform_info"]["dlc_images"] = get_dlc_images(build_context)
+    platform_details["platform_info"]["test_type"] = os.getenv("TEST_TYPE")
+    platform_details["platform_info"]["codebuild_name"] = codebuild_name
+    platform_details["platform_info"]["codebuild_id"] = get_codebuild_project_id()
+    platform_details["platform_info"]["codebuild_url"] = get_target_url(codebuild_name)
+    platform_details["platform_info"]["cloudwatch_logs_url"] = get_cloudwatch_url(codebuild_name)
+    if build_context == "PR":
+        pr_execution_details = get_pr_execution_details()
+        platform_details["platform_info"]["PR"] = pr_execution_details
+    elif build_context == "MAINLINE":
+        mainline_execution_details = get_mainline_execution_details()
+        platform_details["platform_info"]["MAINLINE"] = mainline_execution_details
+
+    return platform_details
+
+
+def get_pr_execution_details():
+    pr_execution_details = {}
+    pr_number = os.getenv("PR_NUMBER")
+    github_url = os.getenv("CODEBUILD_SOURCE_REPO_URL")
+    pr_execution_details["pr_number"] = pr_number
+    pr_execution_details["commit_id"] = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
+    pr_execution_details["github_url"] = github_url
+    pr_execution_details["pr_url"] = f"{github_url}/pull/{pr_number}"
+
+    if config.is_notify_test_failures_enabled:
+        pr_execution_details["notification_severity"] = config.get_notification_severity()
+
+    return pr_execution_details
+
+
+def get_mainline_execution_details():
+    mainline_execution_details = {}
+    codepipeline_name = os.getenv("CODEBUILD_INITIATOR").split("/")[-1]
+    mainline_execution_details["codepipeline_name"] = codepipeline_name
+    mainline_execution_details["codepipeline_execution_id"] = os.getenv("CODEPIPELINE_EXECUTION_ID")
+    mainline_execution_details["code_pipeline_url"] = get_codepipeline_url(codepipeline_name)
+    return mainline_execution_details
+
+
+def parse_pytest_data():
+    """
+    Parse pytest output to get test results.
+    """
+    pytest_raw_data = get_pytest_output()
+    pytest_parsed_output = []
+
+    for f in pytest_raw_data:
```

Comment:
Nit: Might make it more readable to use a better variable for the iterations

Reply:
done